src/components/SlateEditor/plugins/PluginFactory.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Editor, NodeEntry, Element } from "slate";
+import { RenderLeafProps } from "slate-react";
+import config from "../../../config";
+import { NormalizerConfig, defaultBlockNormalizer } from "../utils/defaultNormalizer";
+
+type ElementType = Element["type"];
+
+type MappedPlugins = {
+  [K in ElementType]: Plugin<K>;
+};
+
+interface Plugin<T extends ElementType = ElementType> {
```

Comment:
En tanke: jeg tror det hadde blitt enda mer forståelig ved å legge på en kommentar med beskrivelse av noen av feltene under her, feks `isVoid`, `childPlugins`, `shouldHideBlockPicker`, `renderLeaf` .. 

Gjerne med denne kommentar-syntaxen:   /** Kommentar */, så får man hvertfall i vs-code opp beskrivelsen ved hover

Reply:
Jeg prøvde å legge til noen kommentare, kan hende noen er litt vage men kan jo eventuelt legge til link til info i slate doc 👯

Reply:
Par av kommentarene kan kanskje utbedres:
- isVoid: Denne er upresis. Void elements kan ha children nodes, men forskjellen er om elementet renderes av Slate eller av Elements' render kode. Første setning er good, men kanskje 2. setning istedet burde sagt "A void element is not rendered by Slate, but by the Element's render kode..." og ev. med en henvisning til doc
- renderLeaf: Burde kommentaren istedet reflektert at den er for å overstyre heller enn å gjøre et valg? Dersom renderLeaf ikke er satt, vil default node brukes? Så noe i retning av `Assign to have the leafNode rendered by a specific JSX.element. Otherwise, a default node will be used`